test/unit_test/model/test_pytorch_model.py

```diff
@@ -88,3 +91,85 @@ def test_hf_config_model_class(self):
 
         pytorch_model = olive_model.load_model()
         assert isinstance(pytorch_model, transformers.DistilBertForSequenceClassification)
+
+
+class TestPytorchDummyInput:
+    @pytest.fixture(autouse=True)
+    def setup(self):
+        # hf config values
+        self.task = "text-classification"
+        self.model_class = "DistilBertForSequenceClassification"
+        self.model_name = "distilbert-base-uncased-finetuned-sst-2-english"
+        self.io_config = {
+            "input_names": ["input_ids", "attention_mask", "token_type_ids"],
+            "input_shapes": [[1, 128], [1, 128], [1, 128]],
+            "input_types": ["int64", "int64", "int64"],
+            "output_names": ["output"],
+            "dynamic_axes": {
+                "input_ids": {"0": "batch_size", "1": "seq_length"},
+                "attention_mask": {"0": "batch_size", "1": "seq_length"},
+                "token_type_ids": {"0": "batch_size", "1": "seq_length"},
+            },
+        }
+        self.dataset = {
+            "data_name": "glue",
+            "subset": "mrpc",
+            "split": "validation",
+            "input_cols": ["sentence1", "sentence2"],
+            "label_cols": ["label"],
+            "batch_size": 1,
+        }
+
+    def common_data_config_test(self, olive_model, data_config_template):
+        # mock data config
+        data_config = MagicMock()
+        # mock data container
+        data_container = MagicMock()
+        data_container.get_first_batch.return_value = 1, 0
+        data_config.to_data_container.return_value = data_container
+        # mock data config template
+        data_config_template.return_value = data_config
+
+        # get dummy inputs
+        dummy_inputs = olive_model.get_dummy_inputs()
+
+        data_config_template.assert_called_once()
+        data_config.to_data_container.assert_called_once()
+        data_container.get_first_batch.assert_called_once()
+        assert dummy_inputs == 1
+
+    def test_custom_dummy_inputs(self):
+        dummy_inputs_func = MagicMock(spec=FunctionType)
+        dummy_inputs_func.return_value = 1
+        olive_model = PyTorchModel(
+            hf_config={"task": self.task, "model_name": self.model_name}, dummy_inputs_func=dummy_inputs_func
+        )
+        # get dummy inputs
+        dummy_inputs = olive_model.get_dummy_inputs()
+
+        dummy_inputs_func.assert_called_once_with(olive_model)
+        assert dummy_inputs == 1
+
+    @patch("olive.data.template.dummy_data_config_template")
+    def test_input_shapes_dummy_inputs(self, dummy_data_config_template):
+        olive_model = PyTorchModel(
+            hf_config={"task": self.task, "model_name": self.model_name}, io_config=self.io_config
```

Comment:
it seems the hf_config is not needed?

Reply:
I kept it here since the model needs a source to pass this check https://github.com/microsoft/Olive/blob/main/olive/model/__init__.py#L474. It also indirectly tests the priority order of the dummy inputs.